plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -655,3 +661,27 @@ func (p *Plugin) admitResourceSlice(nodeName string, a admission.Attributes) err
 
 	return nil
 }
+
+func (p *Plugin) admitCSR(nodeName string, a admission.Attributes) error {
+	// The create request must come from a node with the same name in the CN of the CSR.
+	// Other CSR attributes get checked in CSR validation by the signer.
+	if a.GetOperation() == admission.Create {
+		csr, ok := a.GetObject().(*certapi.CertificateSigningRequest)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+		}
+		csrBytes := make([]byte, base64.StdEncoding.DecodedLen(len(csr.Spec.Request)))
+		_, err := base64.StdEncoding.Decode(csrBytes, csr.Spec.Request)
+		if err != nil {
+			return admission.NewForbidden(a, fmt.Errorf("unable to base64 decode csr"))
+		}
+		x509cr, err := certapi.ParseCSR(csrBytes)
+		if err != nil {
+			return admission.NewForbidden(a, fmt.Errorf("unable to parse csr: %v", err))
+		}
+		if strings.HasPrefix(x509cr.Subject.CommonName, "system:node:") && x509cr.Subject.CommonName != fmt.Sprintf("system:node:%s", nodeName) {
+			return admission.NewForbidden(a, fmt.Errorf("can only create a node CSR for the same node name as the requesting node"))
```

Comment:
nit:
```suggestion
			return admission.NewForbidden(a, errors.New("can only create a node CSR for the same node name as the requesting node"))
```

Reply:
errors.New is only useful if you check for that particular error somewhere